contracts/erc721/OpenSeaGasFreeListing.sol

```diff
@@ -36,6 +66,33 @@ library OpenSeaGasFreeListing {
             address(registry) != address(0) &&
             address(registry.proxies(owner)) == operator;
     }
+
+    /**
+    @notice Returns whether the operator is an OpenSea proxy for the owner on a 0x exchange, thus
+    allowing it to list without the token owner paying gas in Polygon mainnet and mumbai.
+    @dev Assumes that the passed in chainId is 137 or 80001.
+     */
+    function isApprovedForAllZeroEx(address operator, uint256 chainId)
+        internal
+        pure
+        returns (bool) 
+    {
+        address registry;
+        assembly {
+            switch chainId
+            case 137 {
+                // polygon
+                registry:= 0x58807baD0B376efc12F5AD86aAc70E78ed67deaE
```

Comment:
(style) Please run `prettier` on the file; there should be a space before the `:=`